### ISSUE TITLE
Problem with Kafka certificates

### ISSUE THREAD

**Issue by codificat** [operate-first/support#1187]
### Describe the Problem

The Kafka brokers in `opf-kafka` are not accepting TLS connections due to some certificate issue.

### Steps to Reproduce

1. Check the logs of a pod which tries to consume from a kafka topic: `investigator-consumer-1-228th` in the `thoth-infra-prod` namespace in smaug
2. See it's trying to connect to `odh-message-bus-kafka-bootstrap.opf-kafka.svc:9093`
3. See it can't connect:

```
SSL handshake failed: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed: broker certificate could not be verified, verify that ssl.ca.location is correctly configured or root CA certificates are installed (install ca-certificates package) (after 11ms in state SSL_HANDSHAKE, 31 identical error(s) suppressed)
```

### Expected behaviour

The consumer should be able to connect to Kafka

### Screenshots

From the logs of one of the kafka brokers (`odh-message-bus-kafka-4` in the `opf-kafka` namespace in smaug):

```
2023-01-08 21:18:12,810 INFO [SocketServer listenerType=ZK_BROKER, nodeId=4] Failed authentication with /10.128.10.33 (SSL handshake failed) (org.apache.kafka.common.network.Selector) [data-plane-kafka-network-thread-4-ListenerName(TLS-9093)-SSL-9]
```

The logs are full of similar entries

### Additional context

Found about this when trying to request a Thoth advise: `thamos advise` just sat there waiting.

In other words: this is preventing Thoth services from functioning normally.

AFAICT, nothing changed recently in the Thoth side of things

**Comment by schwesig** [comment]
https://operatefirst.slack.com/archives/C01RMPVUUK1/p1673266314301049

**Comment by harshad16** [comment]
Strimzi version was update on nov 4, https://console-openshift-console.apps.smaug.na.operate-first.cloud/k8s/ns/opf-kafka/clusterserviceversions/strimzi-cluster-operator.v0.32.0/

with new version of strimzi 0.32.0
it has deprecated the old kafka version like 3.1.x

```
New in 0.32.0
Add support for Kafka 3.3.1 and remove support for Kafka 3.1.0, 3.1.1, and 3.1.2
```
https://github.com/strimzi/strimzi-kafka-operator/releases/tag/0.32.0

As we are at kafka version 3.1.0 
we  have to update it https://github.com/operate-first/apps/blob/72b5175800deb32a43fbb4990efdfec3bd11fa9b/kafka/base/opf-kafka.yaml#L8

**Comment by codificat** [comment]
Many thanks @Gregory-Pereira and @harshad16 for fixing this!

/close
